fido/exceptions.py

```diff
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 import crochet
+import zlib
```

Comment:
If we want to decompress gzip, what's the rationale of using zlib? Can you motivate why not using gzip directly?

Reply:
AFAICT the gzip module for python 2 supports compression and decompression of files only. In addition, we may want to support other formats in the future (e.g deflate or zlib)